crates/derive/src/errors/pipeline.rs

```diff
@@ -17,17 +18,17 @@ macro_rules! ensure {
 }
 
 /// A top level filter for [PipelineError] that sorts by severity.
-#[derive(derive_more::Display, Debug, PartialEq, Eq)]
+#[derive(Error, Debug, PartialEq, Eq)]
 pub enum PipelineErrorKind {
     /// A temporary error.
-    #[display("Temporary error: {_0}")]
-    Temporary(PipelineError),
+    #[error("Temporary error: {0}")]
+    Temporary(#[source] PipelineError),
     /// A critical error.
-    #[display("Critical error: {_0}")]
-    Critical(PipelineError),
+    #[error("Critical error: {0}")]
+    Critical(#[source] PipelineError),
     /// A reset error.
-    #[display("Pipeline reset: {_0}")]
-    Reset(ResetError),
+    #[error("Pipeline reset: {0}")]
+    Reset(#[source] ResetError),
```

Comment:
Can we also remove the `impl From<ResetError>` below and inline with a `#[from]`?

Reply:
Yep! Done